### PR TITLE
Add test cases of naming rules for some interfaces

### LINKED ISSUE #19050: [Bug]: Lack some tests for naming rules
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Lack some tests for naming rules.

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Milvus Log

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by NicoYuan1986**
Resolve: #19050
Signed-off-by: “nico” <cheng.yuan@zilliz.com>

### CODE REVIEW

File: tests/python_client/testcases/test_collection.py [hunk 1]
```diff
@@ -113,6 +113,23 @@ def test_collection_illegal_name(self, name):
         self.collection_wrap.init_collection(name, schema=default_schema, check_task=CheckTasks.err_res,
                                              check_items=error)
 
+    @pytest.mark.tags(CaseLabel.L2)
+    @pytest.mark.parametrize("name", ["_co11ection$","co11_$uper"])
```
Comment (binbinlv) [inline tests/python_client/testcases/test_collection.py:117]: A blank between two values in the list is preferred: ["_co11ection$","co11_$uper"]

File: tests/python_client/testcases/test_index.py [hunk 2]
```diff
@@ -139,6 +139,41 @@ def test_index_params_invalid(self, get_invalid_index_params):
...
+        """
+        target: test index naming rules
+        method: 1. connect milvus
+                2. Create a collection
+                3. Create an index with an index_name which uses all the supported elements in the naming rules
+        expected: Index create successfully
+        """
+        self._connect()
+        collection_w = self.init_collection_wrap()
+        collection_w.create_index(default_field_name, default_index_params, index_name=index_name)
```
Comment (binbinlv) [inline tests/python_client/testcases/test_index.py:154]: Here no check for index name: two ways: 
1. check index name directly after create_index
2. Add check index property check function

File: tests/python_client/testcases/test_search.py [hunk 3]
```diff
@@ -2617,7 +2618,76 @@ def test_search_with_consistency_session(self, nq, dim, auto_id, _async):
...
+    @pytest.mark.xfail(reason="issue 18951")
+    @pytest.mark.parametrize("partition_name", ["_Partii0n", "pArt1_i0n"])
+    def test_search_partition_naming_rules(self, nq, dim, auto_id, partition_name):
+        """
+        target: test search collection naming rules
+        method: 1. Connect milvus
+                2. Create a collection
+                3. Create a partition with a name which uses all the supported elements in the naming rules
+                4. Insert data into collection
+                5. with/without index with a name which uses all the supported elements in the naming rules
```
Comment (binbinlv) [inline tests/python_client/testcases/test_search.py:2672]: This case only covers the search without index, needs another search case with index (index name uses naming rules)

File: tests/python_client/testcases/test_collection.py (hunk 1 above)
Comment (NicoYuan1986) [reply tests/python_client/testcases/test_collection.py]: > A blank between two values in the list is preferred: ["_co11ection$","co11_$uper"]

Added.

File: tests/python_client/testcases/test_index.py (hunk 2 above)
Comment (NicoYuan1986) [reply tests/python_client/testcases/test_index.py:154]: > Here no check for index name: two ways:
> 
> 1. check index name directly after create_index
> 2. Add check index property check function

The check for index name has been added.

File: tests/python_client/testcases/test_search.py (hunk 3 above)
Comment (NicoYuan1986) [reply tests/python_client/testcases/test_search.py]: > This case only covers the search without index, needs another search case with index (index name uses naming rules)

The search case with index (index name uses naming rules) has been added.

File: tests/python_client/testcases/test_search.py [hunk 4]
```diff
@@ -2617,7 +2618,108 @@ def test_search_with_consistency_session(self, nq, dim, auto_id, _async):
                             check_items={"nq": nq,
                                          "ids": insert_ids,
                                          "limit": nb_old + nb_new,
-                                         "_async": _async}) 
+                                         "_async": _async})
+
+    @pytest.mark.tags(CaseLabel.L1)
+    @pytest.mark.parametrize("collection_name", ["_co11ection$", "co11_$uper"])
+    @pytest.mark.parametrize("index_name", ["_1ndeX$", "In_$0"])
```
Comment (binbinlv) [inline tests/python_client/testcases/test_search.py:2625]: Here this parametrize has some problems, will leads multiple collection with same collection name, which is invalid.

Comment (NicoYuan1986) [reply tests/python_client/testcases/test_search.py:2625]: > Here this parametrize has some problems, will create multiple collections with same collection name, which is invalid, need modify.

Done~!😎

**Comment by binbinlv** [comment]
@NicoYuan1986 
Please add the invalid cases for "$".

**Review by binbinlv — COMMENTED** [review]
Please add the invalid case for "$"

**Comment by NicoYuan1986** [comment]
The conflict has been resolved.

**Pull request merged**